Mathlib/Data/Set/Sigma.lean

```diff
@@ -132,6 +133,38 @@ theorem sigma_inter_sigma : s₁.sigma t₁ ∩ s₂.sigma t₂ = (s₁ ∩ s₂
   simp [and_assoc, and_left_comm]
 #align set.sigma_inter_sigma Set.sigma_inter_sigma
 
+theorem _root_.biSup_sigma [CompleteLattice ι'] (s : Set ι) (t : ∀ i, Set (α i))
+    (f : Sigma α → ι') : ⨆ ij ∈ s.sigma t, f ij = ⨆ (i ∈ s) (j ∈ t i), f ⟨i, j⟩ :=
+  by apply eq_of_forall_ge_iff; intro; constructor <;> simp_all
```

Comment:
`aesop` was messing with this theorem statement (i think due to `unhygienic` pulling in implicit variables), so i expanded them all down to `simp`/`simp_all`

Reply:
what do you mean by this? the theorem statement should be fully evaluated before entering tactic mode

Reply:
<img width="753" alt="Screenshot 2024-02-09 at 1 53 53 PM" src="https://github.com/leanprover-community/mathlib4/assets/5601392/9f90c0e8-4e9a-4543-84f4-8270127f9e6f">
<img width="713" alt="Screenshot 2024-02-09 at 1 54 15 PM" src="https://github.com/leanprover-community/mathlib4/assets/5601392/9a85e519-884e-4ba8-a9cf-dc496f850268">

`aesop` introduces `x` as an implicit argument, which is undesired. I don't fully understand the mechanics behind this, but `clear x; aesop` works

Reply:
Oh wow, that's unhelpful, yes. Maybe worth raising this as an aesop issue